### PR TITLE
2098 Use new suggestions in frontend

### LINKED ISSUE #2098: Update front-end to use the new suggestions object
## Subtasks
- [x] Update the Graphql query to get the suggestions entry of the Manuscript
```graphql
-# Something like this
...something extends Manuscript {
   ....
    submission {
        fieldName
        value
    }
}
```
- [x] Write the `suggestion` to the title field in the form (if there's nothing there already)
- [ ] Update the API Test to include the `suggestions` property (even if it's just an empty array) cc @petereast 

blocked by #2034

### PR DESCRIPTION
**Pull request by petereast**
#### Background

What does this PR do?
closes #2098 

#### Any relevant tickets

Please use the phrase 'Closes' if it will fully complete the ticket.

#### How has this been tested?
**Reviewers** ensure that these test requirements are also reviewed.

Please indicate the need for this change to be tested. If not please justify, if so then at what levels. 

- [ ] Unit / Integration tests
- [ ] Browser tests
- [ ] End2end tests

#### UI Changes

- [ ] Has been reviewed against Designs
- [ ] Necessary updates to styleguide

### CODE REVIEW

**Comment by petereast** [comment]
TODO: Make the DetailsStep stateful ( :( ) and do the logic

File: packages/component-submission/client/pages/DetailsStepPage.js [hunk 1]
```diff
@@ -11,122 +12,138 @@ import SubjectAreaDropdown from '../components/SubjectAreaDropdown'
...
-        maxRows={4}
-        name="meta.title"
-        placeholder="Manuscript title"
-      />
-    </Box>
+const useSuggestions = ({ suggestions }) =>
+  suggestions.reduce(
+    (acc, sugg) => ({
+      ...acc,
+      [sugg.fieldName]: last(sortBy(sugg.suggestions, ['score'])).value,
```
Comment (diversemix) [inline packages/component-submission/client/pages/DetailsStepPage.js]: This should probably be sorted by time to select the most recent. Not by score

File: packages/component-submission/client/utils/parseInputToFormData.js [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
-  returnObject.secondCosubmissionTitle =
-    cosubmissionValues.length > 1 ? cosubmissionValues[1] : null
-
-  return returnObject
+  return {
+    firstCosubmissionTitle: cosubmissionValues.length
+      ? cosubmissionValues[0]
+      : null,
+    secondCosubmissionTitle:
+      cosubmissionValues.lengt > 1 ? cosubmissionValues[1] : null,
```
Comment (diversemix) [inline packages/component-submission/client/utils/parseInputToFormData.js]: Length

**Comment by diversemix** [comment]
Should this be marked as WIP

File: packages/component-submission/client/pages/DetailsStepPage.js [hunk 3]
```diff
@@ -11,122 +12,138 @@ import SubjectAreaDropdown from '../components/SubjectAreaDropdown'
...
+    // TODO: turn this ValidatedField into a richTextBox
+    <React.Fragment>
+      <Box mb={3}>
+        <ValidatedField
+          component={ExpandingTextField}
+          data-test-id="manuscript-title-editor"
+          label="Manuscript Title"
+          maxRows={4}
+          name="meta.title"
+          value={title}
```
Comment (Cooryd) [inline packages/component-submission/client/pages/DetailsStepPage.js]: Should this check for a `values.title` value first before populating?

File: packages/component-submission/client/pages/DetailsStepPage.js (hunk 1 above)
Comment (petereast) [reply packages/component-submission/client/pages/DetailsStepPage.js]: `score` at the moment is basically just a sequential index of a suggestion

File: packages/component-submission/client/utils/parseInputToFormData.js (hunk 2 above)
Comment (petereast) [reply packages/component-submission/client/utils/parseInputToFormData.js]: I :heart: JS

**Comment by petereast** [comment]
> Should this be marked as WIP

yes, but not anymore

**Review by diversemix — APPROVED** [review]

**Review by will-byrne — APPROVED** [review]

**Pull request merged**